### PR TITLE
[ENHANCEMENT] Window titles

### PR DESCRIPTION
**Pull request by IdealistCat**
## Does this PR close any issues? If so, link them below.

No

## Briefly describe the issue(s) fixed.

Almost Every state has a Window title (not an issue but a QOL thing)

### CODE REVIEW

**Review by JVNpixels — APPROVED** [review]

**Comment by JVNpixels** [comment]
May be a bit unnecessary to the devs, but I think this is great!

**Comment by IdealistCat** [comment]
I thought it would be cool

**Comment by IdealistCat** [comment]
Thought it would be neat

**Comment by Cartridge-Man** [comment]
ngl this reminds me of leather engine (i literally KNOW that nobody will know what this is at all lol)

File: source/Main.hx [hunk 1]
```diff
@@ -38,6 +39,7 @@ class Main extends Sprite
     // We need to make the crash handler LITERALLY FIRST so nothing EVER gets past it.
     CrashHandler.initialize();
     CrashHandler.queryStatus();
+    WindowUtil.setWindowTitle("Friday Night Funkin'");
```
Comment (NotHyper-474) [inline source/Main.hx:42]: I don't think this is needed as this only ever runs once and game already sets the title to that by default.

**Review by AbnormalPoof — COMMENTED** [review]
I suggest also adding Window Titles for the Animation Editor (`DebugBoundingState`) and the Input Offsets menu (`LatencyState`).

**Comment by IdealistCat** [comment]
> ngl this reminds me of leather engine (i literally KNOW that nobody will know what this is at all lol)

I do

File: source/Main.hx (hunk 1 above)
Comment (IdealistCat) [reply source/Main.hx:42]: ah

**Comment by IdealistCat** [comment]
> I suggest also adding Window Titles for the Animation Editor (`DebugBoundingState`) and the Input Offsets menu (`LatencyState`).

good idea

**Comment by Cartridge-Man** [comment]
>  I do

Good job then ig

**Comment by IdealistCat** [comment]
> > I do
> 
> Good job then ig

:)

File: source/Main.hx (hunk 1 above)
Comment (ayed-jake-Offical) [reply source/Main.hx]: is it like 
`Application.current.window.title = "Friday Night Funkin'";`
or they make a `public function setWindowTitle(titleWindow:String){Application.current.window.title = titleWindow;}` on class WindowUtil
well cool a idea !!!

**Comment by ninjamuffin99** [comment]
i thought about something like this, however can you make sure this is compatible with OBS? I know there's a setting that can match a source to the window title (or same exe type), and I've had troubles with that for Spotify where it forgets and OBS isn't able to detect the window correctly!

**Comment by IdealistCat** [comment]
> i thought about something like this, however can you make sure this is compatible with OBS? I know there's a setting that can match a source to the window title (or same exe type), and I've had troubles with that for Spotify where it forgets and OBS isn't able to detect the window correctly!

No Idea Honestly, never fucked around with OBS Scripting.

**Comment by ninjamuffin99** [comment]
its not so much scripting, but adding the window as a capture source. essentially if someone is playing the game in windowed mode and spitting it out through obs

**Comment by ninjamuffin99** [comment]
I think I will close this one, it's a cool idea but I think *stylistically* we'd just keep the window title nice and simple!

**Comment by IdealistCat** [comment]
> I think I will close this one, it's a cool idea but I think _stylistically_ we'd just keep the window title nice and simple!

understandable

**Pull request closed without merging**